### PR TITLE
Feat/authcontext rewrite to typescript

### PR DESCRIPTION
**Pull request by kajm**
## Explain the changes you’ve made
Re-wrote authactions and authreducer and their functions to typescript.
Also added types.

## Explain why these changes are made
Above mentioned files weren't written in typescript.

## Explain your solution
Created new typescript files
Added interfaces for each action, added two interfaces if the action returned an error, this to differentiate the result from the action.

## How to test

Concrete example:

1. Checkout this branch
2. Fire upp the simulator by running the command `yarn ios` or `yarn android`
3. Run the application as usual
4. 
## Tested environments

- [] Storybook on a iOS device/simulator.
- [x] Building the Application on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-2xqtux5 Skriv om authcontext till typescript](https://app.clickup.com/t/2xqtux5)

File: source/store/actions/AuthActions.ts [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+      type: ActionTypes.signStarted,
+      payload: {
+        orderRef,
+        autoStartToken,
+      },
+    };
+  } catch (error) {
+    return {
+      type: ActionTypes.signFailure,
+      payload: error?.data ?? null,
```
Comment (MasterKenth) [inline source/store/actions/AuthActions.ts:177]: Fix this somehow. `error?.data` _can_ be a `string` (e.g. check `bankid.sign(...)`).

File: source/store/reducers/AuthReducer.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    case ActionTypes.signSuccess:
+      return {
+        ...state,
+        status: "signResolved",
+        orderRef: undefined,
+        autoStartToken: undefined,
+      };
+
+    case ActionTypes.signFailure: {
+      const { payload } = action as { payload: DispatchError };
```
Comment (MasterKenth) [inline source/store/reducers/AuthReducer.ts:140]: (According to other comment) `payload` is possibly a `string` as well.

File: source/store/AppContext.tsx [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
+export interface AppProviderState {
   mode: string;
   handleSetMode(newMode: string): void;
   isDevMode: boolean;
 }
 
+const initialState: AppProviderState = {
+  mode: "",
+  handleSetMode: () => undefined,
+  isDevMode: true,
```
Comment (MasterKenth) [inline source/store/AppContext.tsx:14]: dev mode on by default?

**Review by MasterKenth — CHANGES_REQUESTED** [review]
A few comments; haven't tested everything yet.

File: source/store/actions/AuthActions.ts (hunk 1 above)
Comment (kajm) [reply source/store/actions/AuthActions.ts:177]: added string type

File: source/store/AppContext.tsx (hunk 3 above)
Comment (kajm) [reply source/store/AppContext.tsx:14]: fixed

**Review by MasterKenth — APPROVED** [review]

**Pull request merged**